roles/os_firewall/tasks/iptables.yml

```diff
@@ -4,17 +4,26 @@
   systemd:
     name: firewalld
     state: stopped
-    enabled: no
-    masked: yes
   register: task_result
   failed_when:
     - task_result is failed
     - ('could not' not in task_result.msg|lower)
 
+# These tasks are split in two, as changing masked and enabled status
+# would mutate `changed` field, so it needs to be performed in two tasks
+- name: Mask and disable firewalld service
+  systemd:
+    name: firewalld
+    enabled: no
+    masked: yes
+  when: not task_result.failed_when_result
+
 - name: Wait 10 seconds after disabling firewalld
   pause:
     seconds: 10
-  when: task_result is changed
+  when:
+    - not task_result.failed_when_result
```

Comment:
This will never be false; if failed_when_result is true, then we would have failed and stopped executing by this step.